src/backend/base/langflow/services/utils.py

```diff
@@ -157,17 +164,92 @@ def initialize_session_service() -> None:
+        session: The database session to use for the deletion
+
+    Returns:
+        None
+    """
+    try:
+        # Delete transactions using bulk delete
+        delete_stmt = (
+            delete(TransactionTable)
+            .where(
+                TransactionTable.id.in_(
+                    select(TransactionTable.id)
+                    .order_by(TransactionTable.timestamp.desc())
+                    .offset(settings_service.settings.max_transactions_to_keep)
+                )
+            )
+            .execution_options(synchronize_session="evaluate")
+        )
+
+        await session.exec(delete_stmt)
+        await session.commit()
+        logger.debug("Successfully cleaned up old transactions")
+    except Exception as exc:  # noqa: BLE001
+        logger.error(f"Error cleaning up transactions: {exc!s}")
+        await session.rollback()
+        # Don't re-raise since this is a cleanup task
+
+
+async def clean_vertex_builds(settings_service: "SettingsService", session: AsyncSession) -> None:
+    """Clean up old vertex builds from the database.
+
+    This function deletes vertex builds that exceed the maximum number to keep (configured in settings).
+    It orders vertex builds by timestamp descending and removes the oldest ones beyond the limit.
+
+    Args:
+        settings_service: The settings service containing configuration like max_vertex_builds_to_keep
+        session: The database session to use for the deletion
+
+    Returns:
+        None
+    """
+    try:
+        # Delete vertex builds using bulk delete
+        delete_stmt = (
+            delete(VertexBuildTable)
+            .where(
+                VertexBuildTable.id.in_(
+                    select(VertexBuildTable.id)
+                    .order_by(VertexBuildTable.timestamp.desc())
+                    .offset(settings_service.settings.max_vertex_builds_to_keep)
+                )
+            )
+            .execution_options(synchronize_session="evaluate")
+        )
+
+        await session.exec(delete_stmt)
+        await session.commit()
+        logger.debug("Successfully cleaned up old vertex builds")
+    except Exception as exc:  # noqa: BLE001
+        logger.error(f"Error cleaning up vertex builds: {exc!s}")
+        await session.rollback()
+        # Don't re-raise since this is a cleanup task
+
+
 async def initialize_services(*, fix_migration: bool = False) -> None:
     """Initialize all the services needed."""
     # Test cache connection
     get_service(ServiceType.CACHE_SERVICE, default=CacheServiceFactory())
     # Setup the superuser
     await asyncio.to_thread(initialize_database, fix_migration=fix_migration)
     async with get_db_service().with_async_session() as session:
-        await setup_superuser(get_service(ServiceType.SETTINGS_SERVICE), session)
+        settings_service = get_service(ServiceType.SETTINGS_SERVICE)
```

Comment:
Alignment of the try block seems incorrect.

Reply:
Solved. Thank you!